edit-post/components/header/keyboard-shortcuts-toggle/index.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * WordPress Dependencies
+ */
+import { withSelect, withDispatch } from '@wordpress/data';
+import { displayShortcut } from '@wordpress/keycodes';
+
+/**
+ * WordPress Dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { compose } from '@wordpress/compose';
+import { MenuItem } from '@wordpress/components';
+
+const modalName = 'edit-post/keyboard-shortcut-help';
+
+export function KeyboardShortcutsToggle( { toggleModal, isModalActive } ) {
+	return (
+		<MenuItem
+			icon={ isModalActive && 'yes' }
+			isSelected={ isModalActive }
+			onClick={ toggleModal }
+			shortcut={ displayShortcut.access( 'h' ) }
+		>
+			{ __( 'Keyboard Shortcuts' ) }
+		</MenuItem>
+	);
+}
+
+export default compose( [
+	withSelect( ( select ) => ( {
+		isModalActive: select( 'core/edit-post' ).isModalActive( modalName ),
```

Comment:
I'm thinking it might make more sense to make this a regular button that, when clicked, opens the modal and closes the dropdown menu (using a passed in `onClose` prop). 

This would also fix some weirdness that exists right now where the dropdown menu is dismissed when you click on the shortcuts modal.

![menu](https://user-images.githubusercontent.com/612155/43874286-580a32f8-9bce-11e8-80ad-64ed83d37028.gif)

Reply:
Oh wow - good catch!